apps/navstar_overlay/src/navstar_overlay_lashup_kv_listener.erl

```diff
@@ -106,52 +107,61 @@ terminate(_Reason, _State, _Data) ->
 %%
-%%--------------------------------------------------------------------------------------------------------------
+%%--------------------------------------------------------------------------------
 
 unconfigured(info, EventContent, Data) ->
-    {next_state, applying_first_config, Data, {next_event, internal, EventContent}}.
+    {next_state, configuring, Data, {next_event, internal, EventContent}}.
 
-applying_first_config(internal, 
-            _EventContent = {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Ref}},
-            _Data = #data{ref = Ref}) when is_tuple(Subnet), tuple_size(Subnet) == 2 ->
-    lager:info("Applying first config: ~p~n", [Event]),
-    navstar_overlay_configure:start_link(reply, Event),
+configuring(internal, {lashup_kv_events, NewEvent = #{key := [navstar, overlay, Subnet], ref := Ref}},
+            #data{ref = Ref, events = OldEvent}) when is_tuple(Subnet), tuple_size(Subnet) == 2 ->
+    DeltaEvent = get_delta_event(NewEvent, OldEvent),
+    lager:debug("Applying configuration: ~p", [DeltaEvent]),
+    navstar_overlay_configure:start_link(reply, DeltaEvent),
     keep_state_and_data;
-applying_first_config(info, {navstar_overlay_configure, applied_config}, Data) ->
-    lager:info("Done applying first config"),
-    {next_state, batching, Data};
-applying_first_config(info, _EventContent, _Data) ->
+configuring(info, {navstar_overlay_configure, applied_config, DeltaEvent}, Data0) ->
+    lager:debug("Done applying configuration: ~p", [DeltaEvent]),
+    Data1 = update_state_data(DeltaEvent, Data0),
+    {next_state, batching, Data1};
+configuring(info, _EventContent, _Data) ->
     {keep_state_and_data, postpone}.
             
 batching(info, EventContent, _Data) ->
     {keep_state_and_data, {timeout, ?LISTEN_TIMEOUT, {do_configure, EventContent}}};
 batching(timeout, {do_configure, EventContent}, Data) ->
     {next_state, configuring, Data, {next_event, internal, EventContent}}.
 
-configuring(internal, {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Ref}},
-            Data0 = #data{ref = Ref}) when is_tuple(Subnet), tuple_size(Subnet) == 2 ->
-    Pid = navstar_overlay_configure:start_link(noreply, Event),
-    lager:debug("Starting configurator ~p for config: ~p~n", [Pid, Event]), 
-    Data1 = Data0#data{pid = Pid},
-    {keep_state, Data1};
-configuring(info, _EventContent, Data) ->
-    %% Short circuit the current configurator
-    %% as there is an updated configuration
-    lager:debug("Killing configurator ~p as new config received", [Data#data.pid]), 
-    navstar_overlay_configure:stop(Data#data.pid),
-    {next_state, batching, Data, postpone}.
+%% private API
+
+get_delta_event(NewEvent = #{key := Key, value := NewValue}, OldEvent) ->
+    case maps:get(Key, OldEvent, []) of
+      [] -> 
+          NewEvent;
+      OldValue ->
+          DeltaValue = sets:to_list(sets:subtract(sets:from_list(NewValue), OldValue)),
```

Comment:
Does it make sense to actually do this set work over ordsets? 

Also, can we add a typespec to this function.